src/wampy.js

```diff
@@ -1180,22 +1179,17 @@ class Wampy {
                 return this._log(decodedPayload.err.message);
             }
 
-            argsList = decodedPayload.args;
-            argsDict = decodedPayload.kwargs;
-
-        } else {
-            argsList = data[4];
-            argsDict = data[5];
+            args = decodedPayload.args;
+            kwargs = decodedPayload.kwargs;
         }
 
-        let i = subscription.callbacks.length;
-        while (i--) {
-            await subscription.callbacks[i]({
-                details : options,
-                argsList: argsList,
-                argsDict: argsDict
-            });
+        const callbacksInReverseOrder = [...(subscription.callbacks || [])].reverse();
```

Comment:
There is no practical reason to explicitly reverse the callbacks array. 
`while i--` was used because it is the most simple and performant variant of cycling